packages/cli/src/utils/verifyManifest.ts

```diff
@@ -0,0 +1,97 @@
+
+import { array, number, object, string } from 'zod';
+
+const forbiddenExtensions = ['exe', 'dmg', 'cmd', 'sh', 'bat'];
+const getForbiddenExtensionsErrorMessage = (surfaceName: string) =>
+    `Invalid file extension, \`${surfaceName}.filenameExtension\` can not include "${forbiddenExtensions}".`;
+const AssetCreationShape = object({}).optional();
+
+export const platformAppManifestSchemaV1 = object({
+    appId: string().refine((value) => value.trim() !== '', {
+        message: '`appId` is required',
+    }),
+    appType: string().refine((value) => ['content-block', 'platform-app', 'theme'].includes(value), {
+        message: '`appType` is required',
+    }),
+    surfaces: object({
+        mediaLibrary: object({
+            assetAction: object({
+                type: array(
+                    string().refine((value) =>
+                        ['audio', 'document', 'image', 'video', 'file', 'embeddedContent'].includes(value),
+                    ),
+                ),
+                filenameExtension: array(
+                    string().refine((value) => !forbiddenExtensions.includes(value), {
+                        message: getForbiddenExtensionsErrorMessage('mediaLibrary'),
+                    }),
+                ),
+            }),
+            assetCreation: AssetCreationShape,
+        }).optional(),
+        iconLibrary: object({
+            assetAction: object({
+                type: array(string().refine((value) => ['image'].includes(value))),
+                filenameExtension: array(string().refine((value) => ['svg'].includes(value))),
+            }),
+            assetCreation: AssetCreationShape,
+        }).optional(),
+        logoLibrary: object({
+            assetAction: object({
+                type: array(string().refine((value) => ['image'].includes(value))),
+                filenameExtension: array(
+                    string().refine((value) =>
+                        ['svg', 'jpg', 'jpeg', 'ai', 'eps', 'png', 'tif', 'tiff'].includes(value),
+                    ),
+                ),
+            }),
+            assetCreation: AssetCreationShape,
+        }).optional(),
+        documentLibrary: object({
+            assetAction: object({
+                type: array(
+                    string().refine((value) =>
+                        ['audio', 'document', 'image', 'video', 'file', 'embeddedContent'].includes(value),
+                    ),
+                ),
+                filenameExtension: array(
+                    string().refine((value) => !forbiddenExtensions.includes(value), {
+                        message: getForbiddenExtensionsErrorMessage('documentLibrary'),
+                    }),
+                ),
+            }),
+            assetCreation: AssetCreationShape,
+        }).optional(),
+        workspaceProject: object({
+            assetAction: object({
+                type: array(
+                    string().refine((value) =>
```

Comment:
To have it typed correctly, can you use enums? https://zod.dev/?id=zod-enums
(same for the others array => file extension and type)

Reply:
applied where applicable